src/game/GameStateMenu.cpp

```diff
@@ -215,47 +204,132 @@ void GameStateMenu::handle() {
+                    userInput = USER_DEFAULT;
+                }
+            }
             break;
 
         case SDL_KEYDOWN:
             keyCode = event.key.keysym.sym;
 
             if (keyCode == SDLK_ESCAPE) {
                 play = false;
+                break;
+            }
+
+            //Handling backspace
+            if (hostIPSelected) {
+                if (keyCode == SDLK_BACKSPACE && hostInput.length() > 0) {
+                    hostInput.pop_back();
+                }
+            } else if (usernameSelected) {
+                if (keyCode == SDLK_BACKSPACE && userInput.length() > 0) {
+                    userInput.pop_back();
+                }
+            }
+
+            if (hostIPSelected || usernameSelected) {
+                //Handle copy
+                if (keyCode == SDLK_c && SDL_GetModState() & KMOD_CTRL) {
+                    if (hostIPSelected) {
+                        SDL_SetClipboardText(hostInput.c_str());
+                    } else if (usernameSelected) {
+                        SDL_SetClipboardText(userInput.c_str());
+                    }
+                }
+                //Handle paste
+                else if (keyCode == SDLK_v && SDL_GetModState() & KMOD_CTRL) {
+                    if (hostIPSelected) {
+                        hostInput += SDL_GetClipboardText();
+                    } else if (usernameSelected) {
+                        userInput += SDL_GetClipboardText();
+                    }
+                }
             }
             break;
 
         case SDL_TEXTINPUT:
-        //intentionally left blank for now
-        //only temporary, will be functional later
+            if (hostIPSelected) {
+                //Update the string int displayed in the host IP textbox
+                if (hostInput.length() < maxLength) {
+                    hostInput += event.text.text;
+                }
+            } else if (usernameSelected) {
+                //Update the string int displayed in the username textbox
+                if (userInput.length() < maxLength) {
+                    userInput += event.text.text;
+                }
+            }
             break;
 
         case SDL_KEYUP: //Do nothing on key release
-        //intentionally left blank for now
-        //only temporary, will be functional later
             break;
 
         case SDL_MOUSEMOTION:
-        //intentionally left blank for now
-        //only temporary, will be functional later
+            x = event.motion.x;
+            y = event.motion.y;
+
+            //Checks the mouse position and sets the menu option Join
+            //to it's acive state when the mouse hovers above the Join text
+            if (x >= joinRect.x && x <= joinRect.x + joinRect.w &&
+                    y >= joinRect.y && y <= joinRect.y + joinRect.h) {
+                joinSelected = true;  //Activate the button
+            } else {
+                joinSelected = false; //deactivate the button
```

Comment:
`joinSelected = (x >= joinRect.x && x <= joinRect.x + joinRect.w && y >= joinRect.y && y <= joinRect.y + joinRect.h)`

Unless this will change and you aren't just setting a boolean value